Bottom: 2d980b5f1cdab913f03a89238a2ecbc0afc508c4
Top:    3ac1b32e2ea9332a17cdc3fca5594966d1e49519
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-09 23:48:13 +0000

Refresh of ensure-we-throttle-redraws-fro

---

diff --git a/src/hid/gtk/gui-trackball.c b/src/hid/gtk/gui-trackball.c
index 33b8c81..f5f741a 100644
--- a/src/hid/gtk/gui-trackball.c
+++ b/src/hid/gtk/gui-trackball.c
@@ -76,6 +76,7 @@ button_press_cb (GtkWidget *widget, GdkEventButton *ev, gpointer userdata)
     {
       axis[0] = 1.; axis[1] = 0.; axis[2] = 0.;
       axis_to_quat (axis, 0, ball->quart1);
+      axis_to_quat (axis, 0, ball->quart2);
       gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (ball->view_2d), FALSE);
     }
 
@@ -139,8 +140,10 @@ motion_notify_cb (GtkWidget *widget, GdkEventMotion *ev, gpointer userdata)
   double x2, y2;
   float q[4];
 
-  if (!ball->dragging)
+  if (!ball->dragging) {
+    gdk_event_request_motions (ev);
     return TRUE;
+  }
 
   x1 = ball->x1;
   y1 = ball->y1;
@@ -155,6 +158,7 @@ motion_notify_cb (GtkWidget *widget, GdkEventMotion *ev, gpointer userdata)
   g_signal_emit (ball, ghid_trackball_signals[ROTATION_CHANGED], 0,
                  ball->quart2);
 
+  gdk_event_request_motions (ev);
   return TRUE;
 }
 
@@ -245,7 +249,8 @@ ghid_trackball_constructor (GType type,
 
   gtk_widget_add_events (ball->drawing_area, GDK_BUTTON_PRESS_MASK   |
                                              GDK_BUTTON_RELEASE_MASK |
-                                             GDK_POINTER_MOTION_MASK);
+                                             GDK_POINTER_MOTION_MASK |
+                                             GDK_POINTER_MOTION_HINT_MASK);
 
   return G_OBJECT (ball);
 }
